src/modules.c

```diff
@@ -521,7 +521,7 @@ Function global_table[] = {
   (Function) egg_inet_aton,
   /* 252 - 255 */
   (Function) egg_snprintf,
-  (Function) egg_vsnprintf,
```

Comment:
Why not just export the real vsnprintf here for compatibility?

Reply:
because we already do like:
src/mod/module.h includes src/main.h
src/main.h includes eggdrop.h
and eggdrop.h has it defined for compatibility:
https://github.com/eggheads/eggdrop/pull/1523/files#diff-e717df86c10ed7c4ff2d25b284e548a7054e43d63d74dad305d7ecd5318ac1feR284

Reply:
but you made me thinking, we could as well define some symbols in modules.h instad of eggdrop.h when it is only for mod compatibility.

also the strncpy() define could be streamlined.

just tell me if i shall do all this in this PR, or create a new Issue as a reminder for later

Reply:
ok, done,
now we keep binary mod compatibility
ready for review again.